nixos/modules/services/networking/dnscache.nix

```diff
@@ -86,7 +86,11 @@ in {
 
   config = mkIf config.services.dnscache.enable {
     environment.systemPackages = [ pkgs.djbdns ];
-    users.users.dnscache.isSystemUser = true;
+    users.users.dnscache = {
+        isSystemUser = true;
+        group = "dnscache";
+    };
+    users.groups.dnscache = {};
```

Comment:
Can the service be converted to DynamicUser instead?

Reply:
I don't think so. dnscache wants to bind socket, chroot and drop privileges itself. It has no options to skip any of these steps.